tests/core/fs/test_fs_online_rwfile_and_sync.py

```diff
@@ -13,6 +15,47 @@
 PLAYGROUND_SIZE = BLOCK_SIZE * 10
 
 
+@pytest.mark.trio
+@pytest.mark.slow
+async def test_leak_opened_file(user_fs_online_state_machine, alice: LocalDevice):
+    LOGGER = structlog.getLogger(test_leak_opened_file.__name__)
+    import resource
+
+    resource.setrlimit(resource.RLIMIT_NOFILE, [1024, 1024])
```

Comment:
We must undo this once the test is done, otherwise this may create unexpected side effects in other tests

```python
vanilla_rlimit = resource.getrlimit()
resource.setrlimit(my_rlimit)
try:
    # do the test here
finally:
    resourec.setrlimit(vanilla_rlimit)
```

Reply:
Is there a way to avoid changing the limit ? It would be much better to get the current number of opened file descriptor, this way we wouldn't have to do a 10000 iteration loop, only a couple of cycle to be sure would be enough)

Reply:
I've talked with @vxgmichel about it, I'll try to use [`psutil.Process.open_files`](https://psutil.readthedocs.io/en/latest/#psutil.Process.open_files) & [`psutil.Process.connections`](https://psutil.readthedocs.io/en/latest/#psutil.Process.connections) to compare the number of openned fd between a run.

We can't just increase the iteration loop because it would take a lot of time.
> I've already tested that before settling on `ulimit` but I've had to kill the test after ~1h30.